action/protocol/staking/bucket_index.go

```diff
@@ -68,86 +66,26 @@ func (bis *BucketIndices) deleteBucketIndex(index uint64) {
 	}
 }
 
-func getBucketIndices(sr protocol.StateReader, key []byte) (*BucketIndices, uint64, error) {
-	var bis BucketIndices
-	height, err := sr.State(
-		&bis,
-		protocol.NamespaceOption(StakingNameSpace),
-		protocol.KeyOption(key))
-	if err != nil {
-		return nil, height, err
-	}
-	return &bis, height, nil
-}
-
-func putBucketIndex(sm protocol.StateManager, key []byte, index uint64) error {
-	var bis BucketIndices
-	if _, err := sm.State(
-		&bis,
-		protocol.NamespaceOption(StakingNameSpace),
-		protocol.KeyOption(key)); err != nil && errors.Cause(err) != state.ErrStateNotExist {
-		return err
-	}
-	bis.addBucketIndex(index)
-	_, err := sm.PutState(
-		&bis,
-		protocol.NamespaceOption(StakingNameSpace),
-		protocol.KeyOption(key))
-	return err
-}
-
-func delBucketIndex(sm protocol.StateManager, key []byte, index uint64) error {
-	var bis BucketIndices
-	if _, err := sm.State(
-		&bis,
-		protocol.NamespaceOption(StakingNameSpace),
-		protocol.KeyOption(key)); err != nil {
-		return err
-	}
-	bis.deleteBucketIndex(index)
-
-	var err error
-	if len(bis) == 0 {
-		_, err = sm.DelState(
-			protocol.NamespaceOption(StakingNameSpace),
-			protocol.KeyOption(key))
-	} else {
-		_, err = sm.PutState(
-			&bis,
-			protocol.NamespaceOption(StakingNameSpace),
-			protocol.KeyOption(key))
-	}
-	return err
-}
-
-func getVoterBucketIndices(sr protocol.StateReader, addr address.Address) (*BucketIndices, uint64, error) {
-	return getBucketIndices(sr, addrKeyWithPrefix(addr, _voterIndex))
-}
-
-func putVoterBucketIndex(sm protocol.StateManager, addr address.Address, index uint64) error {
-	return putBucketIndex(sm, addrKeyWithPrefix(addr, _voterIndex), index)
+func (csr *candSR) GetVoterBucketIndices(addr address.Address) (*BucketIndices, uint64, error) {
```

Comment:
rename  Public function`GetVoterBucketIndices` to `VoterBucketIndices`, same for the others